src/praisonai-agents/praisonaiagents/tools/nightmarket_tools.py

```
@@ -0,0 +1,131 @@
+        req_headers = {"Accept": "application/json"}
+        if headers:
+            req_headers.update(headers)
+        if payment_signature:
+            req_headers["payment-signature"] = payment_signature
+
+        data = None
+        if body and method.upper() in ("POST", "PUT", "PATCH"):
+            data = json.dumps(body).encode()
+            req_headers["Content-Type"] = "application/json"
+
+        req = Request(url, data=data, headers=req_headers, method=method.upper())
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)